src/domain/users/index.types.d.ts

```diff
@@ -39,14 +54,22 @@ type User = {
   readonly defaultAccountId: AccountId
   readonly deviceTokens: DeviceToken[]
   readonly createdAt: Date
-  phone: PhoneNumber
+  readonly phone: PhoneNumber
+  readonly twilio: PhoneMetadata | null
   language: UserLanguage
   twoFA: TwoFAForUser
 }
 
+type NewUserInfo = {
+  phone: PhoneNumber
+  twilio?: PhoneMetadata
+}
+
 interface IUsersRepository {
   findById(userId: UserId): Promise<User | RepositoryError>
   findByUsername(username: Username): Promise<User | RepositoryError>
+  findByPhone(phone: PhoneNumber): Promise<User | RepositoryError>
+  persistNew({ phone, twilio }: NewUserInfo): Promise<User | RepositoryError>
```

Comment:
please rename twilio param, repo implementation should take care of the legacy name

Reply:
what would you rename it to?

Reply:
uhmm the type is `PhoneMetadata` so I think is fine to use `metadata` or `phoneMetadata`